### PR TITLE
 Add option to require checker name in warning suppression. (Fixes 2318)

### LINKED ISSUE #2318: Option to make `nullness:` prefix for suppression strings mandatory
As a follow-up to #1701, it might be nice to have a way to make the Checker only recognize suppressions with a prefix corresponding to a type system, e.g. `nullness:`. The advantage of the prefixes is that it makes it possible to recognize suppressions that are no longer needed, as discussed in the other bug.

### PR DESCRIPTION
**Pull request by smillst**
Fixes #2318.

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java [hunk 1]
```diff
@@ -199,6 +199,12 @@
     // org.checkerframework.framework.source.SourceeVisitor.checkForSuppressWarningsAnno
     "warnUnneededSuppressions",
 
+    // Require that warning suppression annotations contain a checker key as a prefix in order for
+    // the warning to be suppressed.
+    // org.checkerframework.framework.source.SourceChecker.checkSuppressWarnings(java.lang.String[],
+    // java.lang.String)
+    "requireCheckerPrefixInWarningSuppressions",
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java]: This name is pretty bad. Maybe just `requireCheckerInWarningSuppressions`?  Other ideas?

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java [hunk 2]
```diff
@@ -199,6 +199,12 @@
     // org.checkerframework.framework.source.SourceeVisitor.checkForSuppressWarningsAnno
     "warnUnneededSuppressions",
 
+    // Require that warning suppression annotations contain a checker key as a prefix in order for
+    // the warning to be suppressed.
+    // org.checkerframework.framework.source.SourceChecker.checkSuppressWarnings(java.lang.String[],
```
Comment (mernst) [inline framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java]: What is the purpose of these two lines of comment?  Is it a cut-and-paste error?

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java (hunk 1 above)
Comment (mernst) [reply framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java]: I think a long name is OK for this rarely-used feature.
I changed it to `requirePrefixInWarningSuppressions`.

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java (hunk 2 above)
Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java]: The comment at the top of the `@SupportedOptions` annotation says:
```
    // When adding a new standard option:
    // 1. Add a brief blurb here about the use case
    //    and a pointer to one prominent use of the option.
```
This is the pointer to the use of the option.

**Pull request merged**